### PR TITLE
Add rollup to build UI, Configure Maven to import UI on compile

### PR DESCRIPTION
**Pull request by eldrickb**
- Adds `rollup` to handle building the UI
- Adds `exec-maven-plugin` to build the UI on compile 
- Configures maven to import `ui/dist` as webapp resources to `target`
  - With this move to import the UI, `server/src/../webapp` has been deleted
- Removes a couple files included by scaffold tools that aren't needed

With these changes, the project is pretty much fully scaffolded. The only thing missing is Jasmine testing (which I assumed needed to have time dedicated to it to add to the project but based on our pod sync that sounds like it probably isn't really the case)

### CODE REVIEW

File: server/pom.xml [hunk 1]
```diff
@@ -1,58 +1,94 @@
...
+<project
+	xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>com.google.sps</groupId>
+	<artifactId>Voxetta</artifactId>
+	<version>1</version>
+	<packaging>war</packaging>
+	<properties>
+		<!-- Configured for 1.8 (for now); compiled on 11 -->
```
Comment (ashleygranitto) [inline server/pom.xml:11]: What impact (if any) will the configuration versus compiling version difference make?

File: server/pom.xml [hunk 2]
```diff
@@ -1,58 +1,94 @@
...
+			<artifactId>javax.servlet-api</artifactId>
+			<version>4.0.1</version>
+			<scope>provided</scope>
+		</dependency>
+		<dependency>
+			<groupId>com.google.code.gson</groupId>
+			<artifactId>gson</artifactId>
+			<version>2.8.6</version>
+		</dependency>
+		<!-- TODO(eldrickb): is this the correct version? -->
```
Comment (ashleygranitto) [inline server/pom.xml:35]: this is the version I used for my portfolio website

File: server/pom.xml [hunk 3]
```diff
@@ -1,58 +1,94 @@
...
+		<dependency>
+			<groupId>com.google.appengine</groupId>
+			<artifactId>appengine-api-1.0-sdk</artifactId>
+			<version>1.9.59</version>
+		</dependency>
+	</dependencies>
+	<build>
+		<!-- TODO(eldrickb): import UI on build -->
```
Comment (mmallett) [inline server/pom.xml:43]: This is done with this pull request?

File: server/pom.xml [hunk 4]
```diff
@@ -1,58 +1,94 @@
...
+		<plugins>
+			<plugin>
+				<groupId>com.google.cloud.tools</groupId>
+				<artifactId>appengine-maven-plugin</artifactId>
+				<version>2.2.0</version>
+				<configuration>
+					<!-- TODO: set project ID. -->
```
Comment (mmallett) [inline server/pom.xml:50]: Who is the owner of this TODO

File: ui/rollup.config.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import merge from 'deepmerge';
+import resolve from '@rollup/plugin-node-resolve';
+import copy from 'rollup-plugin-copy';
+
+// use createSpaConfig for bundling a Single Page App
+import {createSpaConfig} from '@open-wc/building-rollup';
+// use createBasicConfig to do regular JS to JS bundling
+// import { createBasicConfig } from '@open-wc/building-rollup';
```
Comment (mmallett) [inline ui/rollup.config.js:8]: remove commented out code

File: ui/rollup.config.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+// TODO: make SPA decision and cleanup this config further
```
Comment (mmallett) [inline ui/rollup.config.js:10]: Assign an owner for TODO

Comment (win-chen) [reply ui/rollup.config.js:10]: Would it be better to assign an issue number to the TODO? And have a list of open issues as trackers?

File: server/pom.xml (hunk 1 above)
Comment (eldrickb) [reply server/pom.xml:11]: honestly, no idea

File: server/pom.xml (hunk 2 above)
Comment (eldrickb) [reply server/pom.xml:35]: yeah but some things from the portfolio POM were also out of date making me question, I'll re-look into this later as it works for now

File: ui/rollup.config.js (hunk 6 above)
Comment (eldrickb) [reply ui/rollup.config.js]: assigned to myself for now, not sure if github does automated issue creation/tracking but I know a todo bot exists

**Pull request merged**